### DISCUSSION TITLE
Agenda 2023-01-17

### DISCUSSION THREAD

**Discussion by kelset** [microsoft/rnx-kit#2110] [Bundle Working Group]
* bye bye Adam Foxman
  * does this meeting needs to change format?
* coming up: align-deps + rnx-kit conversation (https://github.com/microsoft/rnx-kit/issues/1863)
* dynamic requires convo
  * inline requires via Babel plugin: https://github.com/facebook/fbjs/blob/main/packages/babel-preset-fbjs/plugins/inline-requires.js (is it the same as the transform flag in Metro config?)
  * *"although it changes the semantics of the JS a bit in some edge cases, so it isn't totttttally safe"*
  *  doesn't really work for ES Modules (?)
  * https://github.com/facebook/metro/issues/682
    * https://github.com/react-native-async-storage/async-storage/pull/896
  * it only works with `require` and not `import` (there's a fix but not in OSS)
  * https://github.com/EvanBacon/Metro-Context-Modules-Demo
* *"Optional deps don’t really work. We should organize and fix in 2023"* (Evan) https://github.com/facebook/metro/issues/836
* how's Meta planning shaping up?
* TS types for Metro? https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/metro
* SWC and other plans for transformers? (exotic mode)

**Comment by motiz88** [comment]
Re inline requires:

> is [the fbjs plugin] the same as the transform flag in Metro config?

Yes, it is: https://github.com/facebook/metro/blob/5c6897eba2ac6d5524f554f11f5baaa5d3156e43/packages/metro-transform-worker/src/index.js#L325

**Comment by cortinico** [comment]
> coming up: align-deps + rnx-kit conversation (https://github.com/microsoft/rnx-kit/issues/1863)

I've missed this as I haven't got the meeting invite. Anything relevant?
